src/input.rs

```diff
@@ -466,18 +492,49 @@ impl<'a, A: ActionContext + 'a> Processor<'a, A> {
     }
 
     pub fn mouse_input(&mut self, state: ElementState, button: MouseButton, modifiers: ModifiersState) {
-        if let MouseButton::Left = button {
-            let state = mem::replace(&mut self.ctx.mouse_mut().left_button_state, state);
-            if self.ctx.mouse_mut().left_button_state != state {
-                match self.ctx.mouse_mut().left_button_state {
-                    ElementState::Pressed => {
-                        self.on_mouse_press(modifiers);
-                    },
-                    ElementState::Released => {
-                        self.on_mouse_release(modifiers);
+        match button {
+            MouseButton::Left => {
+                let state = mem::replace(&mut self.ctx.mouse_mut().left_button_state, state);
+                if self.ctx.mouse_mut().left_button_state != state {
+                    match self.ctx.mouse_mut().left_button_state {
+                        ElementState::Pressed => {
+                            self.on_mouse_press(button, modifiers);
+                        },
+                        ElementState::Released => {
+                            self.on_mouse_release(button, modifiers);
+                        }
+                    }
+                }
+            }
+            MouseButton::Middle => {
+                let state = mem::replace(&mut self.ctx.mouse_mut().middle_button_state, state);
+                if self.ctx.mouse_mut().middle_button_state != state {
+                    match self.ctx.mouse_mut().middle_button_state {
+                        ElementState::Pressed => {
+                            self.on_mouse_press(button, modifiers);
+                        },
+                        ElementState::Released => {
+                            self.on_mouse_release(button, modifiers);
+                        }
                     }
                 }
             }
+            MouseButton::Right => {
+                let state = mem::replace(&mut self.ctx.mouse_mut().right_button_state, state);
+                if self.ctx.mouse_mut().right_button_state != state {
+                    match self.ctx.mouse_mut().right_button_state {
+                        ElementState::Pressed => {
+                            self.on_mouse_press(button, modifiers);
+                        },
+                        ElementState::Released => {
+                            self.on_mouse_release(button, modifiers);
+                        }
+                    }
+                }
+            }
+            MouseButton::Other(_) => {
+                // Handling more than three buttons properly is impossible.
+            }
```

Comment:
Why is that impossible? I don't think we need more than three buttons, but it seems possible to me.

Also changing this to
```rust
// Handling more than three buttons properly is impossible.
MouseButton::Other(_) => (),
```
would be the better choice, I think.

Reply:
Can fix the comment; as for handling more than three buttons:
* VT200 (i.e., what alacritty calls "normal" mode) has no concept of more than three buttons plus two wheel directions.
* SGR mode lumps all buttons past the third into a single button code of "3".
* VT200 mode uses button code "3" for button-release events.
* Both VT200 and SGR mode use button code 3 to represent "no button pressed" for any-event motion tracking mode.

With all of those, you can properly support *one* extra button, only in SGR mode, and only if you don't use any-event motion tracking. That's close enough to impossible for me.